app/services/AFTPartialService.scala

```diff
@@ -433,15 +368,16 @@ class AFTPartialService @Inject()(
     (implicit messages: Messages): DashboardAftViewModel = {
 
     val subHeadingPaymentDue = {
+
       val upcomingCharges: Seq[PsaFS] =
-        paymentsAndChargesService.extractUpcomingCharges[PsaFS](psaFs, _.dueDate)
+        psaFs.filter(charge => charge.dueDate.nonEmpty && !charge.dueDate.get.isBefore(DateHelper.today))
```

Comment:
Perhaps better to use charge.dueDate.exists(_.isBefore(DateHelper.today))) - more concise

Reply:
Updated